### PR TITLE
docs: Clarify that RegexMatcher uses a full regex match

### PR DESCRIPTION
**Pull request by ggreenway**
Signed-off-by: Greg Greenway <ggreenway@apple.com>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/type/matcher/v3/regex.proto [hunk 1]
```diff
@@ -62,7 +62,8 @@ message RegexMatcher {
     GoogleRE2 google_re2 = 1 [(validate.rules).message = {required: true}];
   }
 
-  // The regex match string. The string must be supported by the configured engine.
+  // The regex match string. The string must be supported by the configured engine. The regex should
```
Comment (ggreenway) [inline api/envoy/type/matcher/v3/regex.proto:65]: I'm not thrilled with this wording. Anyone have a better way to phrase it?

Comment (daixiang0) [reply api/envoy/type/matcher/v3/regex.proto:65]: `The string supported by the specified regex engine`?

Comment (markdroth) [reply api/envoy/type/matcher/v3/regex.proto:65]: How about "The regex is matched against the full string, not as a partial match."

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
Let's just wait for green CI.

**Pull request merged**